### PR TITLE
[Merged by Bors] - docs(Tactic/Rename): improve the docstring

### PR DESCRIPTION
**Pull request by urkud**

### CODE REVIEW

File: Mathlib/Tactic/Rename.lean [hunk 1]
```diff
@@ -11,7 +11,9 @@ open Lean Elab.Tactic Meta
 
 syntax renameArg := term " => " ident
 
-/-- `rename' h => hnew` renames the hypothesis named `h` to `hnew`. -/
+/-- `rename' h => hnew` renames the hypothesis named `h` to `hnew`.
+To rename several hypothesis, use `rename' h₁ => h₁new, h₂ => h₂new`.
+You can use `rename' a => b, b => a` to swap two variables. -/
```
Comment (gebner) [inline Mathlib/Tactic/Rename.lean:16]: Wait, this works?!?  That was certainly not the intention (or the way the code is written).

**Comment by semorrison** [comment]
Just noticed this peculiar behaviour:
```
import Mathlib.Tactic.Rename

example (a : Nat) : true := by
  rename' a => b
  -- Goal is now `true = true`!
  rfl
```

File: Mathlib/Tactic/Rename.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Tactic/Rename.lean:16]: This does work; I added a test case.

Comment (gebner) [reply Mathlib/Tactic/Rename.lean:16]: Okay, then my hint was too subtle.  It doesn't work in half of the cases:
```lean
example (b : Nat) (a : Int) : Nat × Int := by
  rename' a => b, b => a
  exact (a, b)
```

Comment (semorrison) [reply Mathlib/Tactic/Rename.lean:16]: Oops. :-)

**Comment by gebner** [comment]
> Just noticed this peculiar behaviour:

Check out the goal state *before* `rename'` to be even more amazed!  :smile: 

In Lean 4, lowercase `true` is `Bool.true` (formely known as `bool.tt`).  Coercing it to a proposition adds the `= true`.  The true proposition is called uppercase `True`.

**Comment by semorrison** [comment]
> > Just noticed this peculiar behaviour:
> 
> Check out the goal state _before_ `rename'` to be even more amazed! 😄
> 
> In Lean 4, lowercase `true` is `Bool.true` (formely known as `bool.tt`). Coercing it to a proposition adds the `= true`. The true proposition is called uppercase `True`.

Gah!

**Pull request closed without merging**